File: DarkOrbit 10.0/Net/netty/commands/PetGearAddCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class PetGearAddCommand
    {
        public const short ID = 28351;

        public static byte[] write(PetGearTypeModule gearType, int level, int amount, bool enabled)
        {
            var param1 = new ByteArray(ID);
            param1.writeInt(amount >> 10 | amount << 22);
            param1.writeShort(17438);
            param1.writeBoolean(enabled);
            param1.writeInt(level >> 2 | level << 30);
            param1.write(gearType.write());
            return param1.ToByteArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/PetGearAddCommand.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

/**
Created by LEJYONER on 21/09/2017.
*/

public class PetGearAddCommand
        implements ServerCommand {

    public static int ID       = 16710;
    public        PetGearTypeModule gearType;
    public        int level  = 0;
    public 		  int amount = 0;
    public 	boolean enabled = false;

    public PetGearAddCommand(PetGearTypeModule param1, int param2, int param3, boolean param4) {
        this.gearType = param1;
        this.level = param2;
        this.amount = param3;
        this.enabled = param4;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
        	this.gearType.write(param1);
            param1.writeShort(-2528);
            param1.writeInt(this.amount >>> 5 | this.amount << 27);
            param1.writeBoolean(this.enabled);
            param1.writeInt(this.level << 4 | this.level >>> 28);
            } catch (IOException e) {
        }
    }
}